js_modules/dagit/packages/ui/src/components/MiddleTruncate.tsx

```diff
@@ -28,88 +52,65 @@ export const MiddleTruncate: React.FC<Props> = (props) => {
     [text],
   );
 
-  React.useLayoutEffect(() => {
-    if (ref.current) {
-      const width = ref.current.getBoundingClientRect().width;
-      setTargetStyle({font: getFont(ref.current), width});
-    }
-  }, []);
-
-  const debouncedObserver = React.useMemo(() => {
-    return debounce((entry: ResizeObserverEntry) => {
-      setTargetStyle((current) => {
-        const {width} = entry.contentRect;
-        if (current) {
-          return {...current, width};
-        }
-        if (ref.current) {
-          return {font: getFont(ref.current), width};
-        }
-        return null;
-      });
-    }, 10);
-  }, []);
-
-  useResizeObserver(ref.current, debouncedObserver);
-
   return (
-    <MiddleTruncateDiv ref={ref} onCopy={handleCopy} title={text}>
-      {truncated}
-    </MiddleTruncateDiv>
+    <Container onCopy={handleCopy} title={text}>
+      <MeasureWidth ref={measure}>{text}</MeasureWidth>
+      <span>{truncatedText}</span>
+    </Container>
   );
-};
+});
 
-const MiddleTruncateDiv = styled.div`
+// An invisible target element that contains the full, no-wrapped text. This is used
+// to measure the maximum available width for our truncated string.
+const MeasureWidth = styled.div`
+  height: 0;
```

Comment:
Zero height seems to work well enough for getting the necessary width, without a `visibility: hidden`.